src/zenml/zen_server/utils.py

```diff
@@ -570,3 +575,63 @@ def verify_admin_status_if_no_rbac(
                 "without RBAC enabled.",
             )
     return
+
+
+def is_user_request(request: Request) -> bool:
+    """Determine if the incoming request is a user request.
+
+    This function checks various aspects of the request to determine
+    if it's a user-initiated request or a system request.
+
+    Args:
+        request (Request): The incoming FastAPI request object.
+
+    Returns:
+        bool: True if it's a user request, False otherwise.
+    """
+    # Define system paths that should be excluded
+    system_paths: List[str] = [
+        "/health",
+        "/metrics",
+        "/system",
+        "/docs",
+        "/redoc",
+        "/openapi.json",
+    ]
+
+    # Check if the path starts with the user endpoint prefix
+    user_prefix = f"{API}{VERSION_1}{ACTIONS}"
+    if request.url.path.startswith(user_prefix):
+        return True
```

Comment:
What is this for ?

Reply:
so all user requests go through /api/v1/.. however i had ACTIONS in there by mistake. fixed it now

Reply:
I think the question is, why do we need this, if in this case none of the below will be met, so True will be returned anyways.

Reply:
i just wnated to be sure somehow ... but maybe I was overthinking it